informer/spec.go

```diff
@@ -0,0 +1,19 @@
+package informer
+
+import (
+	"k8s.io/client-go/pkg/runtime"
+)
+
+// Observer functions are called when the cache.ListWatch List and Watch
+// functions are called.
+type Observer interface {
+	OnList()
+	OnWatch()
+}
+
+// ZeroObjectFuncs provides zero values of an object and objects' list ready to
+// be decoded. The provided zero values must not be reused by zeroObjectFactory.
+type ZeroObjectFactory interface {
+	NewObject() runtime.Object
+	NewObjectList() runtime.Object
+}
```

Comment:
I don't see a reason of creating this spec.go file. None of these interfaces are foundation of this package.  IMO `ZeroObectFactory` belongs to `zero_object.go` and `Observer` should be gone (see my other comment).

The meat of this package is `New` function not these two interfaces.